### DISCUSSION TITLE
Trying to query output type via fragment doesn't work when output is interface

### DISCUSSION THREAD

**Discussion by DoumanAsh** [apollographql/router#924] [Implementation]
I didn't raise issue because I need time to prepare sample.

We basically have mutation that returns interface as output instead of concrete type.
If we use fragment to query it, it seems we just get empty output from router (it happens with router as standalone binary and embedded router)

I wonder if anyone noticed bug with fragments before?
Query works if you remove fragment and just inline its content inside query.

**Comment by bnjjj** [comment]
Would be happy to have a small reproductible example to be able to fix it. Thanks

**Comment by DoumanAsh** [comment]
@bnjjj I extracted my code into public space
It is wrapper over `apollo-router-core` which provides simplified interface with ability to use `async-graphql` schema inside embedded router https://github.com/DoumanAsh/graphql-router-example

We see the same issue with latest preview of `apollo-router` as separate binary too

Here is my test example as PR https://github.com/DoumanAsh/graphql-router-example/pull/1

It is present on branch: `broken_fragment`

Revision of `apollo-router-core` is `86e77965b536bfded97f6ef9687096eb2c520026`

Here is supergraph schema generated by Rover https://github.com/DoumanAsh/graphql-router-example/blob/broken_fragment/tests/supergraph.graphql

Here is `UserType` interface alongside concrete impementations:
https://github.com/DoumanAsh/graphql-router-example/blob/broken_fragment/tests/user.graphql#L5-L10=
https://github.com/DoumanAsh/graphql-router-example/blob/broken_fragment/tests/user.graphql#L15-L17=

My query:
https://github.com/DoumanAsh/graphql-router-example/blob/broken_fragment/tests/router.rs#L343-L354=

```
    fragment TypeBase on UserType {
      __typename
      description
    }
    query Query {
      meType {
        ...TypeBase
      }
    }
```

Its result is empty:
```
{
  "data": {
    "meType": {
       }
    }
}
```

But if I inline fragment into query I get proper output:
```
{
  "data": {
     "meType": {
       "__typename": "RegularUser",
       "description":"regular"
     }
    }
}
```

**Reply by DoumanAsh** [reply]
I added log to my subgraph implementation to see what query comes from supergraph:

```
query Query__user__0{meType{__typename ...on RegularUser{__typename description}...on SuperUser{__typename description}}}
```

In case of inlined fragment seems to be the same

```
query Query__user__0{meType{__typename ...on RegularUser{__typename description}...on SuperUser{__typename description}}}
````

**Reply by DoumanAsh** [reply]
In case you're in doubt about my code.
My subgraph correctly returns:

```
{"data":{"meType":{"__typename":"RegularUser","description":"regular"}}}
```

**Reply by DoumanAsh** [reply]
Let's continue this discussion over issue https://github.com/apollographql/router/issues/928 since someone already raised it
